tests/integration/_experimental/dataset/test_dataset.py

```diff
@@ -148,17 +149,18 @@ def test__register_dataset__composite() -> None:
         for i in range(1, 20)
     ]
     columns = datapoints[0].keys()
+    id_fields = ["id"]
```

Comment:
I haven't done thorough testing but id_fields appear to work for composite style keys. I'd recommend we use "a.text" and "b.text" here instead of top level id field

Reply:
i think there could be discussion on that, and will leave that to your PR if we do want infer id even for composite. Technically the test setup here is correct.

Reply:
Main reason I pointed this out, is we currently have no test coverage for composite style id_fields

Reply:
updated as suggested.